odoo/tools/misc.py

```diff
@@ -1196,6 +1198,72 @@ def format_date(env, value, lang_code=False, date_format=False):
 
     return babel.dates.format_date(value, format=date_format, locale=locale)
 
+
+def format_datetime(env, value, tz=False, dt_format='medium', lang_code=False):
+    """ Formats the datetime in a given format.
+
+     : param {str, datetime} value: datetime to format either in string or in datetime
+    """
+    if not value:
+        return ''
+    if isinstance(value, str):
+        timestamp = odoo.fields.Datetime.from_string(value)
+    else:
+        timestamp = value
+
+    tz_name = tz or env.user.sudo().tz or 'UTC'
```

Comment:
Side question: isn't the user on environment already sudo-ed ? We could gain 20 000 queries ...

Reply:
Yes indeed ! So this is not required !